docs: add docs on GPU support (#1174)
edgelesssys/contrast: docs/docs/getting-started/bare-metal.md

Increase the `user.max_inotify_instances` sysctl limit by adding `user.max_inotify_instances=8192` to `/etc/sysctl.d/99-sysctl.conf` and running `sysctl --system`.

## K3s setup

1. Follow the [K3s setup instructions](https://docs.k3s.io/) to create a cluster.
2. Install a block storage provider such as [Longhorn](https://docs.k3s.io/storage#setting-up-longhorn) and mark it as the default storage class.

> ## Preparing a cluster for GPU usage

Reviewer: imo this sec could use some sub-sections

:::warning
Currently, Contrast only supports GPU workloads on SEV-SNP-based clusters.

> :::

Reviewer: would suggest to use tabs here as well and put the warning on the tdx tab.

To enable GPU usage on a Contrast cluster, some conditions need to be fulfilled for *each cluster node* that should host GPU workloads:

> 1. You must activate the IOMMU. You can check by running `ls /sys/kernel/iommu_groups`. If the output contains the group indices (`0`, `1`, ...), the IOMMU is supported on the host.

Reviewer: No commands inline, please use a code fence.

> 1. The IOMMU needs to be activated. This can be checked by running `ls /sys/kernel/iommu_groups`. If the output contains the group indices (`0`, `1`, ...), the IOMMU is supported on the host.
>    If it doesn't, `intel_iommu=on` needs to be added to the kernel command line.

Reviewer 2: ```suggestion
1. You must activate the IOMMU. You can check by running `ls /sys/kernel/iommu_groups`. If the output contains the group indices (`0`, `1`, ...), the IOMMU is supported on the host.
   Otherwise, add `intel_iommu=on` to the kernel command line.
```
prefer active voice

2. Additionally, the host kernel needs to have the following kernel configuration options enabled:
    - `CONFIG_VFIO`
    - `CONFIG_VFIO_IOMMU_TYPE1`
    - `CONFIG_VFIO_MDEV`
    - `CONFIG_VFIO_MDEV_DEVICE`
    - `CONFIG_VFIO_PCI`

> If the per-node requirements are fulfilled, deploy the NVIDIA GPU Operator to the cluster. It provisions pod-VMs with GPUs via VFIO.

Reviewer: please add an upstream link for nvidia gpu operator

> If the per-node requirements are fulfilled, the NVIDIA GPU Operator needs to be deployed to the cluster. It manages the facilitation of pod-VMs with GPUs via VFIO.

Reviewer 2: ```suggestion
If the per-node requirements are fulfilled, deploy the NVIDIA GPU Operator to the cluster. It provisions pod-VMs with GPUs via VFIO.
```
You often use the word "facilitate" and I have no idea what it's supposed to mean :D

Initially, label all nodes that *should run GPU workloads*:

```sh
kubectl label node <node-name> nvidia.com/gpu.workload.config=vm-passthrough
```

> For a general setup, the operator can then be deployed like so:

Reviewer 2: ```suggestion
For a general setup, you can then deploy the operator with the following command:
```

```sh
helm install --wait --generate-name \
   -n gpu-operator --create-namespace \
   nvidia/gpu-operator \
   --version=v24.9.1 \
   --set sandboxWorkloads.enabled=true \
   --set sandboxWorkloads.defaultWorkload='vm-passthrough' \
   --set nfd.nodefeaturerules=true \
   --set vfioManager.enabled=true \
   --set kataManager.enabled=true \
   --set ccManager.enabled=true
```

> These settings might be tweaked specific to the use-case. For example, `ccManager.enabled` (the confidential computing manager daemonset) is superfluous if no CC-capable GPUs like the NVIDIA H100 are available for the given deployment.

Reviewer 2: ```suggestion
These settings might be tweaked specific to the use-case. For example, `ccManager.enabled` (the confidential computing manager DaemonSet) is superfluous if no CC-capable GPUs like the NVIDIA H100 are available for the given deployment.
```

The [official installation instructions](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html) can be consulted as a guideline of what options are required for the specific use-case.

> For any GPU-enabled Contrast deployment, the minimal set of required options is the aforementioned one, excluding `ccManager.enabled`.

Reviewer 3: I'm not sure if we should just keep it simple and ignore the case of connecting non CC-GPUs to Contrast.
Might be confusing?

Author: I think 99% of people don't have access to H100 GPUs, even in "professional" setups. What do you think @thomasten?

Reviewer 2: If I understand correctly, the command above also works for non CC-GPUs, it just unnecessarily deploys the CC manager. I'd then shorten this section to something like "See the official installation instructions for details and further options."
